intake_sql/intake_sql.py

```diff
@@ -224,6 +249,18 @@ def to_dask(self):
         self._get_schema()
         return self._dataframe
 
+    def to_ibis(self):
+        """
+        Create an ibis expression for the data source.
+        The sql_expr for the source must be a table, not a table expression.
+        The ibis expression is not partitioned.
+        """
+        client = make_ibis_client(self._uri)
+        if self._sql_expr not in client.list_tables():
+            raise ValueError("Only full tables can be used in to_ibis")
```

Comment:
I suppose the table expression might be indeed a whole table (or view) - can that be found out, or do you wish to strictly keep to string table names?

Reply:
This is a limitation of the ibis SQLAlchemy-based clients. Some backends have been set up to use an expression as a subquery, but not these.

Here is an example of a client without such a restriction: https://github.com/Quansight/intake-omnisci/blob/089beb0de75d021221791cc49796ed8e8ac20456/intake_omnisci/source.py#L124-L127